### PR TITLE
NaN is supported by core

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

For historical reason, `Nan` was considered an illegal value. But Realm Core supports the value for floating-point properties.

## ☑️ ToDos

* [x] 📝 Changelog entry
* ~[ ] 📝 `Compatibility` label is updated or copied from previous entry~
* [x] 🚦 Tests
* ~[ ] 📝 Public documentation PR created or is not necessary~
* ~[ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
* ~[ ] typescript definitions file is updated~
* ~[ ] jsdoc files updated~
* ~[ ] Chrome debug API is updated if API is available on React Native~

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,8 +4,7 @@ x.x.x Release notes (yyyy-MM-dd)
 * None.
 
 ### Fixed
-* <How to hit and notice issue? what was the impact?> ([#????](https://github.com/realm/realm-js/issues/????), since v?.?.?)
-* None.
+* Not-a-number (NaN) was wrongly assumed not to be an illegal values in Realm Core (thanks to @jsflax for discovering this).
```
Comment (bmunkholm) [inline CHANGELOG.md]:
```suggestion
* Not-a-number (NaN) was wrongly assumed to be an illegal values in Realm Core (thanks to @jsflax for discovering this).
```

File: src/jsc/jsc_value.hpp [hunk 2]
```diff
@@ -177,8 +177,7 @@ inline double jsc::Value::to_number(JSContextRef ctx, const JSValueRef &value) {
         throw jsc::Exception(ctx, exception);
     }
     if (isnan(number)) {
-        throw std::invalid_argument(util::format("Value '%1' not convertible to a number.",
-                                                 (std::string)to_string(ctx, value)));
+        std::nan("");
```
Comment (fealebenpae) [inline src/jsc/jsc_value.hpp]: return?

Comment (bmunkholm) [inline src/jsc/jsc_value.hpp]:
```suggestion
        return std::nan("");
```

Comment (bmunkholm) [reply src/jsc/jsc_value.hpp]: Seems like we should add some tests that would have detected this?

Comment (tgoyne) [reply src/jsc/jsc_value.hpp]: Isn't `if (isnan(number) { return std::nan(""); } return number;` just a roundabout way of writing `return number`? When does the extra check do anything?

Comment (kneth) [reply src/jsc/jsc_value.hpp]: I think you are right. But `NaN` can be represented by multiple bit patterns in IEEE 754. Maybe I am just seeing ghosts where no ghost exist.

Comment (kneth) [reply src/jsc/jsc_value.hpp]: I think the RN tests will detect it. Locally, I only ran the test using node.js.

**Comment by kneth** [comment]
It seems to work now expect when debugging in React Native. I believe the reason is than JSON doesn't support nan as a number, so we cannot move nan between the app on the device and the debugger in Chrome.

File: src/jsc/jsc_value.hpp (hunk 2 above)
Comment (bmunkholm) [reply src/jsc/jsc_value.hpp]: I mean, `std::nan("")` doesn't do anything... You at least need to return the value.
But as TG writes, why not remove it altogether - unless you can show with a test that it actually does something?

**Comment by kneth** [comment]
I don't think we will be able to support NaN before we support Flipper for RN debugging.

**Pull request closed without merging**